unittests/integration/qubit_allocation.cpp

```diff
@@ -35,7 +35,7 @@ std::vector<cudaq::complex> randomState(int numQubits) {
 
 struct test_state_vector_init {
   void operator()(const std::vector<cudaq::complex> &stateVec) __qpu__ {
-    cudaq::qvector q(cudaq::state{stateVec});
+    cudaq::qvector q(stateVec);
```

Comment:
Might be good to leave at least one of these as an explicit call to `cudaq::state` ctor.

Reply:
I added this new test case in https://github.com/NVIDIA/cuda-quantum/pull/3647/commits/237aa9bf9f11c675c0f4dd14c5e941f72b376c18